openedx/core/djangoapps/credentials/utils.py

```diff
@@ -28,6 +29,13 @@ def get_user_credentials(user):
         log.warning('Credentials configuration is disabled.')
         return credentials
 
+    use_cache = credentials_config.is_cache_enabled
+    if use_cache:
+        cache_key = credentials_config.CACHE_KEY + '.' + user.username
```

Comment:
Since this is caching the result of calling `get_user_credentials` I would say it has to cache per-user since the result is user-specific.  But, it is reasonable to ask how expensive this caching is going to be, and what performance difference it will make.  @ahsan-ul-haq what are your expectations here?